packages/gatsby/src/query-runner/query-compiler.js

```diff
@@ -50,7 +50,8 @@ class Runner {
   }
 
   async parseEverything() {
-    let files = await globp(`${this.baseDir}/**/*.js`)
+    let files = await globp(`${this.baseDir}/**/*.+(t|j)s?(x)`)
+    files = files.filter(d => !d.match(/(\.stories\.|\.test\.|\.d\.)/))
```

Comment:
This line is only to avoid a warning on .d.ts file in my current project, I've just add stories and tests to optimize a little bit more ;) (so it's not really important for now if any test file are not filtered)